tests/dotnet/UnitTests/TestBaseClass.cs

```diff
@@ -15,6 +16,17 @@ public abstract class TestBaseClass {
 			{ "_BundlerVerbosity", "1" },
 		};
 
+		readonly char [] invalidChars = { '{', '}', '(', ')', '$', ':', ';', '\"', '\'', ',', '=', '|' };
+		protected string TestName {
+			get {
+				var result = TestContext.CurrentContext.Test.Name;
+				foreach (var c in invalidChars.Concat (Path.GetInvalidPathChars ().Concat (Path.GetInvalidFileNameChars ()))) {
+					result = result.Replace (c, '_');
+				}
+				return result.Replace ("_", string.Empty);
```

Comment:
You can write a lot cleaner code using a Split with several chars and a Join with (_):
```csharp
var invalid = invalidChars.Concat (Path.GetInvalidPathChars ().Concat (Path.GetInvalidFileNameChars ()));
result = string.Join ("_", result.Split(separators, StringSplitOptions.RemoveEmptyEntries));
```
I know it is just a test, but you are creating a new string for every iteration for every character. We can be better in performance.